### PR TITLE
KAFKA-9798: Send one round synchronously before starting the async producer

### PR DESCRIPTION
**Pull request by guozhangwang**
Comparing all other test cases, the `shouldAllowConcurrentAccesses` starts an async producer sending records throughout the test other than just synchronously sent and acked a few records before we start the streams application. Right after the streams app is started, we check that at least one record is sent to the output topic (i.e. completed processing). However since only this test starts the producer async and did not wait for it to complete, it is possible that the async producer gets too longer to produce some records and causing it to fail.

To follow what other tests did, I let this test to first send one round of records synchronously before starting the async producing.

Also encountered some new scala warnings that I fixed along with this PR.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@mjsax @cadonna to take a look.

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java [hunk 1]
```diff
@@ -1262,6 +1266,7 @@ public void run() {
                     }
                     incrementIteration();
                 }
+                producer.flush();
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java]: The producer is create using try-with-resource clause thus, the producer should be flushed on the implicit `close()` anyway? Thus, I don't think explicit flushing would help?

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java [hunk 2]
```diff
@@ -631,7 +631,11 @@ public void shouldAllowConcurrentAccesses() throws Exception {
         final String storeName = "word-count-store";
         final String windowStoreName = "windowed-word-count-store";
 
-        final ProducerRunnable producerRunnable = new ProducerRunnable(streamConcurrent, inputValues, numIterations);
+        // send one round of records first to populate the stores
+        ProducerRunnable producerRunnable = new ProducerRunnable(streamThree, inputValues, 1);
+        producerRunnable.run();
+
+        producerRunnable = new ProducerRunnable(streamConcurrent, inputValues, numIterations);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java]: Should this be `numIterations - 1`

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java]: Makes sense, will fix.

**Comment by guozhangwang** [comment]
@mjsax thanks for the review! Addressed your comments.

**Review by mjsax — APPROVED** [review]
I don't think this will fix the test, but maybe help narrow it down. Feel free to merge.

**Pull request merged**